framework/src/org/checkerframework/framework/type/DefaultTypeHierarchy.java

```diff
@@ -779,6 +780,27 @@ public Boolean visitWildcard_Typevar(AnnotatedWildcardType subtype, AnnotatedTyp
 
     @Override
     public Boolean visitWildcard_Wildcard(AnnotatedWildcardType subtype, AnnotatedWildcardType supertype, VisitHistory visited) {
+        // This method is called at a method invocation where a type variable in the method
+        // declaration is substituted with a wildcard.
+        // For example:
+        // <T> void method(Gen<T> t) {}
+        // Gen<?> x;
+        // method(x); // this method is called when checking this method call
+
+        if (subtype.getUnderlyingType() == supertype.getUnderlyingType()) {
+            boolean subtypeHasAnno = subtype.getAnnotationInHierarchy(currentTop) != null;
+            boolean supertypeHasAnno = supertype.getAnnotationInHierarchy(currentTop) != null;
+
+            if (subtypeHasAnno && supertypeHasAnno) {
+                // if both have primary annotations then just check the primary annotations
+                // as the bounds are the same
+                return isPrimarySubtype(subtype, supertype, true);
+
+            } else if (!subtypeHasAnno && !supertypeHasAnno && areEqualInHierarchy(subtype, supertype, currentTop)) {
+                // two unannotated uses of the same wildcard are of the same type
```

Comment:
I don't think this is true in general. Two different wildcard uses might bind to two different types.
I think a big TODO to correctly implement capture conversion would be good here.